### DISCUSSION TITLE
How do I install required dependencies?

### DISCUSSION THREAD

**Discussion by moshegottlieb** [SwiftPackageIndex/SwiftPackageIndex-Server#1637] [Q&A] [answered]
So, my package requires the development package of sqlite3 installed.  
It's a part of the Apple SDKs, so no problem there, but for Linux, that's not the case.  
I've added two providers to the `Package.swift` file:
```swift
providers: [
            .apt(["libsqlite3-dev"]),
            .yum(["sqlite-devel"])])
)
```
Unfortunately, the swift package manager doesn't use it at the time being, so `swift build` doesn't do anything with it.  
How do I tell your build system I need a package?

**Comment by daveverwer** [comment, marked as answer]
Hi Moshe!

We run all our Linux builds through Docker, and we do allow custom Docker images for cases like this. I thought we had documented this in the [builder FAQ](https://swiftpackageindex.com/docs/builds), but it seems that we did not. 😬 The best documentation there is the original PR that enabled this functionality, in #686.

We need to whitelist Docker images before our build infrastructure can build using them, and you'll need to make images for each version of Swift that you're compatible with based on the corresponding base Swift image.

Then, add the `.spi.yml` file and let us know what the images are so we can vet them, and it should all magically work! 😅

**Reply by moshegottlieb** [reply]
Thanks, sounds too much trouble, plus I'm going to have to maintain those dockers as swift continues to evolve.  
I'm just going to add some linux specific instructions in my README.

**Reply by finestructure** [reply]
Actually you're in luck, because we need sqlite for SPI itself. So you can just reference our base image and get it all for free :)

```yml
version: 1
builder:
  configs:
  - platform: linux
    swift_version: '5.5'
    image: registry.gitlab.com/finestructure/spi-base:0.6.4
```

We've also got a little more documentation in the launch blog post here (but it'd be nice to have it available centrally for sure):

https://blog.swiftpackageindex.com/posts/the-swift-package-index-metadata-file-first-steps/
